test/Microsoft.AspNetCore.Server.KestrelTests/MemoryPoolIteratorTests.cs

```diff
@@ -381,5 +381,66 @@ public void GetsKnownVersion(string input, char endChar, bool expectedResult, st
 
             _pool.Return(block);
         }
+
+        [Theory]
+        [InlineData("HTTP/1.0\r", "HTTP/1.0")]
+        [InlineData("HTTP/1.1\r", "HTTP/1.1")]
+        public void KnownVersionsAreInterned(string input, string expected)
+        {
+            TestKnownStringsInterning(input, expected,
+                (iter, expect) => {
```

Comment:
I think there's a way to make this simpler. You could declare a delegate like

    delegate bool GetKnownString(MemoryPoolIterator iter, out string result);

Then have `TestKnownStringsInterning` take a `GetKnownString` and call it and check against it's `expected` param. That way you don't need this lambda here and IMO the test looks clearer.

To be honest though, we already check for that the output is correct in `GetsKnownMethod` and `GetsKnownVersion`, so this check is actually redundant. You're already making the desired check for interning in `TestKnownStringsInterning`.